scripts/tools/RegisterPublishedDeliveryWebhook.php

```diff
@@ -0,0 +1,131 @@
+
+class RegisterPublishedDeliveryWebhook extends ScriptAction
+{
+    use ServiceManagerAwareTrait;
+
+    private const DEFAULT_HTTP_METHOD = 'GET';
+
+    const ACCEPTED_HTTP_METHOD = [
+        'POST',
+        self::DEFAULT_HTTP_METHOD
+    ];
+    /** @var Report */
+    private $report;
+
+    protected function provideOptions()
+    {
+        return [
+            'simpleRosterUrl' => [
+                'prefix' => 'u',
+                'longPrefix' => 'url',
+                'flag' => false,
+                'description' => 'Will register endpoint to inform about new deliveries published',
+                'required' => true
+            ],
+            'httpMethod' => [
+                'prefix' => 'm',
+                'description' => 'Determine what type of http method use',
+                'flag' => false,
+                'required' => false
+            ]
+        ];
+    }
+
+    protected function provideDescription()
+    {
+        return 'Script will register webhook with a defined url';
+    }
+
+    protected function run(): Report
+    {
+        $this->report = Report::createInfo('Registering webhook');
+
+        /** @var ConfigurableService $eventManager */
+        $eventManager = $this->getServiceLocator()->get(EventManager::class);
+        /** @var ConfigurableService $webhookEventsService */
+        $webhookEventsService = $this->getServiceLocator()->get(WebhookEventsService::class);
+
+        try {
+            $webhookEventsService->registerEvent(RemoteDeliveryCreatedEvent::class);
+            $this->getServiceManager()->register(WebhookEventsService::SERVICE_ID, $webhookEventsService);
+            $this->getServiceManager()->register(EventManager::SERVICE_ID, $eventManager);
+        } catch (Exception $exception) {
+            $this->report->add(
+                Report::createError('Registering failed')
+            );
+        }
+
+        $this->getWebhookRegistryManager()->addWebhookConfig(
+            $this->createWebHook(),
+            RemoteDeliveryCreatedEvent::class
+        );
+
+        return $this->report;
+    }
+
+    private function getWebhookRegistryManager(): WebhookRegistryManagerInterface
+    {
+        return $this->getServiceLocator()->get(WebhookRegistryManager::class);
+    }
+
+    private function getHttpMethod(): string
+    {
+        $method = $this->getOption('httpMethod') ?? self::DEFAULT_HTTP_METHOD;
+        if (!in_array($method, self::ACCEPTED_HTTP_METHOD)) {
+            $this->report->add(
+                Report::createWarning(sprintf("Used illegal method %s. Falling back to default method", $method))
+            );
+            return self::DEFAULT_HTTP_METHOD;
+        }
+
+        return $method;
+    }
+
+    private function createWebHook()
+    {
+        return new Webhook(
+            'remoteDeliveryWebHook',
+            $this->getOption('simpleRosterUrl'),
+            $this->getHttpMethod(),
+            5,
```

Comment:
can we name this parameter, because the int 5 does not explain what we are trying to achieve